python/across_api/base/api.py

```diff
@@ -34,3 +39,37 @@ async def epoch(
 
 
 EpochDep = Annotated[datetime, Depends(epoch)]
+
+
+def daterange(
+    begin: Annotated[
+        datetime,
+        Query(description="Start time of period to be calculated.", title="Begin"),
+    ],
+    end: Annotated[
+        datetime, Query(description="End time of period to be calculated.", title="End")
+    ],
+) -> dict:
+    """
+    Helper function to convert begin and end to datetime objects.
+    """
+    return {"begin": Time(begin), "end": Time(end)}
+
+
+DateRangeDep = Annotated[dict, Depends(daterange)]
+
+
+def stepsize(
+    stepsize: Annotated[
+        int,
+        Query(
+            ge=1,
+            title="Step Size",
+            description="Time resolution in which to calculate result in seconds.",
+        ),
+    ] = 60,
+) -> int:
+    return stepsize * u.s
+
+
+StepSizeDep = Annotated[int, Depends(stepsize)]
```

Comment:
I'm confused on why these belong in `api.py`? Should we have a separate file for dependencies? 

Only routed endpoints should exist in our `api.py` files. Maybe even move the `app` instantiation to a separate file as well.

Reply:
I'm only using dependency injection in the API endpoints, not in the main code. This is why they live in `api.py`, it just seemed like a natural fit for anything API related. I'm not against moving them elsewhere, perhaps to `depends.py`?

FYI: Originally this code had an Ephem API endpoint, but I took it out. This `StepSizeDep` is a remnant of that API endpoint, so I just removed it in 35fc4c9.